src/hooks/useUsers/index.tsx

```diff
@@ -1,20 +1,78 @@
-import { useSuspenseQuery } from '@tanstack/react-query';
+import { URLS } from '@/config/urls';
+import { UserRequest } from '@/types/User';
+import { Axios } from '@/utils/Axios';
+import {
+  useMutation,
+  useQueryClient,
+  useSuspenseQuery
+} from '@tanstack/react-query';
 
 import { usersQueryOptions } from './usersQueryOptions';
 
 const useUsers = () => {
+  const queryClient = useQueryClient();
+
   const {
     data: users,
     refetch: usersRefetch,
     isLoading: usersLoading,
     isError: usersError
   } = useSuspenseQuery(usersQueryOptions);
 
+  const {
+    mutateAsync: createUser,
+    isPending: createUserPending,
+    isError: createUserError
+  } = useMutation({
+    mutationFn: (userData: UserRequest): Promise<UserRequest> =>
+      Axios.post(URLS.CREATE_USER(), userData).then(res => {
+        queryClient.invalidateQueries({ queryKey: ['users'] });
+        return res.data;
+      })
+  });
+
+  const {
+    mutateAsync: editUser,
+    isPending: editUserPending,
+    isError: editUserError
+  } = useMutation({
+    mutationFn: (userData: UserRequest): Promise<UserRequest> =>
+      Axios.put(URLS.EDIT_USER(userData.id || ''), userData).then(res => {
+        queryClient.invalidateQueries({ queryKey: ['users'] });
+        return res.data;
+      })
+  });
+
+  const {
+    mutateAsync: deleteUser,
+    isPending: deleteUserPending,
+    isError: deleteUserError
+     } = useMutation({
+    mutationFn: (userData: UserRequest): Promise<UserRequest> =>
```

Comment:
This shouldnt take whole UserRequest object if all it uses is id. It should only take the id as an argument.

`mutationFn: (id: string): Promise<void> =>`